exchanges/okx/okx.go

```diff
@@ -0,0 +1,4312 @@
+			return nil, errMissingInstrumentID
+		}
+		if args[x].ClientSuppliedOrderID == "" && args[x].OrderID == "" {
+			return nil, errMissingClientOrderIDOrOrderID
+		}
+		if args[x].NewQuantity < 0 && args[x].NewPrice < 0 {
+			return nil, errInvalidNewSizeOrPriceInformation
+		}
+	}
+	var resp []OrderData
+	return resp, ok.SendHTTPRequest(ctx, exchange.RestSpot, amendMultipleOrdersEPL, http.MethodPost, amendBatchOrders, &args, &resp, true)
+}
+
+// ClosePositions Close all positions of an instrument via a market order.
+func (ok *Okx) ClosePositions(ctx context.Context, arg *ClosePositionsRequestParams) (*ClosePositionResponse, error) {
+	if arg.InstrumentID == "" {
+		return nil, errMissingInstrumentID
+	}
+	if arg.MarginMode != "" &&
+		arg.MarginMode != TradeModeCross &&
+		arg.MarginMode != TradeModeIsolated {
+		return nil, errMissingMarginMode
+	}
+	var resp []ClosePositionResponse
+	err := ok.SendHTTPRequest(ctx, exchange.RestSpot, closePositionEPL, http.MethodPost, closePositionPath, arg, &resp, true)
+	if err != nil {
+		return nil, err
+	}
+	if len(resp) == 1 {
+		return &resp[0], nil
+	}
+	return nil, errNoValidResponseFromServer
+}
+
+// GetOrderDetail retrieves order details given instrument id and order identification
+func (ok *Okx) GetOrderDetail(ctx context.Context, arg *OrderDetailRequestParam) (*OrderDetail, error) {
+	if arg == nil {
+		return nil, errNilArgument
+	}
+	params := url.Values{}
+	if arg.InstrumentID == "" {
+		return nil, errMissingInstrumentID
+	}
+	params.Set("instId", arg.InstrumentID)
+	switch {
+	case arg.OrderID == "" && arg.ClientSupplierOrderID == "":
+		return nil, errMissingClientOrderIDOrOrderID
+	case arg.ClientSupplierOrderID == "":
+		params.Set("ordId", arg.OrderID)
+	default:
+		params.Set("clOrdId", arg.ClientSupplierOrderID)
+	}
+	var resp []OrderDetail
+	err := ok.SendHTTPRequest(ctx, exchange.RestSpot, getOrderDetEPL, http.MethodGet, common.EncodeURLValues(tradeOrder, params), nil, &resp, true)
+	if err != nil {
+		return nil, err
+	}
+	if len(resp) == 1 {
+		return &resp[0], nil
+	}
+	return nil, errNoValidResponseFromServer
+}
+
+// GetOrderList retrieves all incomplete orders under the current account.
+func (ok *Okx) GetOrderList(ctx context.Context, arg *OrderListRequestParams) ([]OrderDetail, error) {
+	if arg == nil {
+		return nil, errNilArgument
+	}
+	params := url.Values{}
+	if arg.InstrumentType != "" {
+		params.Set("instType", strings.ToUpper(arg.InstrumentType))
+	}
+	if arg.InstrumentID != "" {
+		params.Set("instId", arg.InstrumentID)
+	}
+	if arg.Underlying != "" {
+		params.Set("uly", arg.Underlying)
+	}
+	if arg.OrderType != "" {
+		params.Set("orderType", strings.ToLower(arg.OrderType))
+	}
+	if arg.State != "" {
+		params.Set("state", arg.State)
+	}
+	if !arg.Before.IsZero() {
+		params.Set("before", strconv.FormatInt(arg.Before.UnixMilli(), 10))
+	}
+	if !arg.After.IsZero() {
+		params.Set("after", strconv.FormatInt(arg.After.UnixMilli(), 10))
+	}
```

Comment:
```suggestion
	if arg.Before != "" {
		params.Set("before", arg.Before)
	}
	if arg.After != "" {
		params.Set("after", arg.After)
	}
```
Order list does not support timestamp based filtering